src/lib/realmdb/__tests__/UserProfileStorage.js

```diff
@@ -0,0 +1,455 @@
+    expect(decrypted).toEqual({})
+  })
+
+  it('should decrypt invalid type profile', async () => {
+    const decrypted = await userProfileStorage._decryptProfileFields(false)
+    expect(decrypted).toEqual({})
+  })
+
+  it('should set multiple profile fields', async () => {
+    const oldProfile = userProfileStorage.profile
+    const fieldsToUpdate = {
+      fullName: {
+        value: 'John Doe',
+        display: 'John Doe',
+        privacy: 'public',
+      },
+      username: {
+        value: 'johndoe123',
+        display: 'johndoe123',
+        privacy: 'public',
+      },
+    }
+    await userProfileStorage.setProfileFields(fieldsToUpdate)
+    const newProfile = userProfileStorage.profile
+
+    expect(newProfile).not.toEqual(oldProfile)
+    Object.keys(fieldsToUpdate).forEach(key => {
+      expect(fieldsToUpdate[key].value).toEqual(newProfile[key].value)
+    })
+
+    // Reset profile
+    await userProfileStorage.setProfile(profileKeyValue, true)
+  })
+
+  it('should set private profile field', async () => {
+    const oldProfile = userProfileStorage.profile
+    const username = 'johndoe123'
+    await userProfileStorage.setProfileField('username', username, 'private')
+    const newProfile = userProfileStorage.profile
+
+    expect(newProfile.username).toEqual({
+      display: '******',
+      value: username,
+      privacy: 'private',
+    })
+    expect(newProfile.username).not.toEqual(oldProfile.username)
+  })
+
+  it('should set public profile field', async () => {
+    const oldProfile = userProfileStorage.profile
+    const username = 'johndoe321'
+    await userProfileStorage.setProfileField('username', username)
+    const newProfile = userProfileStorage.profile
+
+    expect(newProfile.username).toEqual({
+      display: username,
+      value: username,
+      privacy: 'public',
+    })
+    expect(newProfile.username).not.toEqual(oldProfile.username)
+  })
+
+  it('should throw error for invalid privacy setting', async () => {
+    try {
+      await userProfileStorage.setProfileField('username', 'johndoe1111', '123123123')
+    } catch (exception) {
+      expect(exception.message).toEqual('Invalid privacy setting')
+    }
+  })
+
+  it('should get profile by wallet address', async () => {
+    // Reset profile
+    await userProfileStorage.setProfile(profile, true)
+    const foundProfile = await userProfileStorage.getProfileByWalletAddress(
+      userProfileStorage.profile.walletAddress.display,
+    )
+    const usersProfile = await userProfileStorage.profiledb.getProfile()
+
+    // Check if correct profile was found
+    expect(foundProfile.user_id).toEqual(usersProfile.user_id)
+  })
+
+  it('should not find a wallet with invalid address', async () => {
+    const foundProfile = await userProfileStorage.getProfileByWalletAddress('123123123')
+    expect(foundProfile).toBeNull()
+  })
+
+  it('should get profile by email', async () => {
```

Comment:
all of these get profile by tests  are redundant and doesnt add any new scenario